### PR TITLE
Switch to ubuntu-20.04 to fix gcov issue

### PR DESCRIPTION
**Pull request by ret2libc**
Reading a bit online, it seems some solved the gcov issue by using a newer gcc. So this PR moves our github CI to ubuntu-20.04. THis change was anyway going to happen soon because github CI is moving ubuntu-latest from 18.04 to 20.04.

It seems it actually makes the gcov job much faster and I don't see the problem about "merge function 0".

### CODE REVIEW

**Comment by XVilka** [comment]
Please also test Windows with AppVeyor.

**Comment by ret2libc** [comment]
> Please also test Windows with AppVeyor.

It's not necessary. I've only touched files that are not considered by appveyor

**Review by XVilka — APPROVED** [review]
It's great.
Interestingly, there seem to be a variable size bug somewhere in the debugging code:
```
0x00000000
0x00400590
0x00000003
0x00400590
0xfffffffffffff30e
0x00000000
0x0040052f
```

**Comment by ret2libc** [comment]
> It's great.

It is! It makes linux-meson-gcc job much faster!

> Interestingly, there seem to be a variable size bug somewhere in the debugging code:
> 
> ```
> 0x00000000
> 0x00400590
> 0x00000003
> 0x00400590
> 0xfffffffffffff30e
> 0x00000000
> 0x0040052f
> ```

Yeah... I have just "fixed" the test for now because I wanted to see how the jobs behaved, but this should be fixed, if it is an actual problem.

**Review by kazarmy — APPROVED** [review]
Why is this still in draft?

**Comment by ret2libc** [comment]
> Why is this still in draft?

I need to *really* fix the debug tests.

**Comment by kazarmy** [comment]
Don't be surprised though to find that each debug test fix is an adventure in itself.

File: test/db/archos/linux-x64/dbg_step_back [hunk 1]
```diff
@@ -14,7 +14,7 @@ dk 9
 EOF
 EXPECT=<<EOF
 0x00000001
-0x00000001
+0x736c6974756572
```
Comment (ret2libc) [inline test/db/archos/linux-x64/dbg_step_back:17]: checked this on my machine, fedora 32, and this value is 0. So i assume it depends on the system. The other values remain the same. Maybe a better test would be to just check rbx, r12 and rip.

File: test/db/archos/linux-x64/dbg_step_back [hunk 2]
```diff
@@ -37,16 +37,16 @@ dk 9
 EOF
 EXPECT=<<EOF
 0x00400574
-0x00000000
 0x00400590
-0x00000003
 0x00400590
+0xfffffffffffff30e
```
Comment (ret2libc) [inline test/db/archos/linux-x64/dbg_step_back:42]: these values are simply not stable, as they may contain whatever data... so I'd say it is alright to just adjust the tests. In the long term we probably want to have better way to test debug tests.

**Pull request merged**